File: NickLodewijks/src/org/uva/sea/ql/ast/expr/NEq.java
package org.uva.sea.ql.ast.expr;

public class NEq extends BooleanExpr {
	private final Expr lhs;
	private final Expr rhs;

	public NEq(Expr lhs, Expr rhs) {
		this.lhs = lhs;
		this.rhs = rhs;
	}

	@Override
	public Boolean interpret(Context context) {
		return !lhs.interpret(context).equals(rhs.interpret(context));
	}
}

File: Edward Poot/src/ql/ast/expr/binaryoperatorexpr/neq.go
package binaryoperatorexpr

import "ql/ast/expr"

type NEq struct {
	BinaryOperator
}

func NewNEq(lhs expr.Expr, rhs expr.Expr) NEq {
	return NEq{BinaryOperator{Lhs: lhs, Rhs: rhs}}
}

func (n NEq) Eval(s interface{}) interface{} {
	switch n.Lhs.Eval(s).(type) {
	case int:
		return n.Lhs.Eval(s).(int) != n.Rhs.Eval(s).(int)
	case bool:
		return n.Lhs.Eval(s).(bool) != n.Rhs.Eval(s).(bool)
	default:
		panic("NEq error: comparing unknown types")
	}
}
